sunpy/map/mapbase.py

```diff
@@ -678,8 +678,17 @@ def spatial_units(self):
         """
         Image coordinate units along the x and y axes (i.e. cunit1, cunit2).
         """
-        return SpatialPair(u.Unit(self.meta.get('cunit1', 'arcsec')),
-                           u.Unit(self.meta.get('cunit2', 'arcsec')))
+        cunit1 = self.meta.get('cunit1')
+        cunit2 = self.meta.get('cunit2')
+        msg = ('Image coordinate units for axis {} not present in metadata. '
+               'It can be set using `map.meta["cunit{}"] = unit`.')
+        if cunit1 is None:
+            raise AttributeError(msg.format('1', '1'))
+        if cunit2 is None:
+            raise AttributeError(msg.format('2', '2'))
```

Comment:
```python
cunits = {f'cunit{i}: self.meta.get(f'cunit{i}') for i in [1, 2]}
msg = ('Image coordinate units for axis {} not present in metadata. '
       'It can be set using `map.meta["{}"] = unit`.')
for lab in cunits:
    if cunits[cunit_i] is None:
         raise AttributeError(msg.format(cunit_i[-1], cunit_i))
```

Also, the problem I see here is that you need to run it once, fix `cunit1`, then encounter other error, run it again...  What about:

```python
err_message = []
for lab in cunits:
    if cunits[cunit_i] is None:
         err_message.append(msg.format(cunit_i[-1], cunit_i))

if err_message:
    raise AttributeError('\n'.join(err_message))

```